### PR TITLE
Removed misleading list of tested OS in GEOS API docs.

### PR DESCRIPTION
**Pull request by ajosephau**
#### Trac ticket number

Not applicable

#### Branch description
Reference: https://discord.com/channels/856567261900832808/859997770274045954/1301540597068202039

#### Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [x] I have checked the "Has patch" ticket flag in the Trac system.
- [x] I have added or updated relevant tests.
- [x] I have added or updated relevant docs, including release notes if applicable.
- [x] I have attached screenshots in both light and dark modes for any UI changes.

### CODE REVIEW

File: docs/ref/contrib/gis/geos.txt [hunk 1]
```diff
@@ -34,8 +34,7 @@ features include:
   may be used outside of a Django project/application.  In other words,
   no need to have :envvar:`DJANGO_SETTINGS_MODULE` set or use a database, etc.
 * Mutability: :class:`GEOSGeometry` objects may be modified.
-* Cross-platform and tested; compatible with Windows, Linux, Solaris, and
-  macOS platforms.
+* Cross-platform and tested.
```
Comment (smithdc1) [inline docs/ref/contrib/gis/geos.txt:37]: Thanks for picking this up. 👍

Maybe remove the "and"?

```suggestion
* Cross-platform tested.
```

Comment (ulgens) [inline docs/ref/contrib/gis/geos.txt:37]: Wouldn't it be better to remove Solaris? I think the new version of that sentence hides information unnecessarily.

Comment (felixxm) [reply docs/ref/contrib/gis/geos.txt:37]: We don't test on MacOS or Solaris, and on Windows testing is quite limited. IMO, there is no point in making false promises here.

Comment (ulgens) [reply docs/ref/contrib/gis/geos.txt:37]: Got it. I didn't realize it's the case. Do we have a clean list of testing combinations?

Comment (felixxm) [reply docs/ref/contrib/gis/geos.txt:37]: https://code.djangoproject.com/wiki/CI
djangoci.com

**Review by claudep — APPROVED** [review]
:+1:

**Review by felixxm — APPROVED** [review]

**Pull request merged**